### PR TITLE
[service.subtitles.traduttorianonimi] 2.0.3

### PR DESCRIPTION
**Pull request by ShellAddicted**

### CODE REVIEW

File: service.subtitles.traduttorianonimi/addon.xml [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+    <description lang="en">Find and download subtitles from www.traduttorianonimi.it</description>
+    <summary lang="it">servizio sottotitoli TraduttoriAnonimi</summary>
+    <description lang="it">Trova e scarica sottotitoli da www.traduttorianonimi.it</description>
+	<language>it</language>
+    <platform>all</platform>
+    <license>GNU GENERAL PUBLIC LICENSE. Version 3, June 2007</license>
+    <forum></forum>
+    <website>http://www.traduttorianonimi.it</website>
+    <email>shelladdicted@gmail.com</email>
+    <source></source>
```
Comment (learningit) [inline service.subtitles.traduttorianonimi/addon.xml]: Please add <source> and also, if you can, <forum> links.

File: service.subtitles.traduttorianonimi/service.py [hunk 2]
```diff
@@ -0,0 +1,184 @@
...
+
+__addon__ = xbmcaddon.Addon();
+__author__ = __addon__.getAddonInfo("author");
+__scriptid__ = __addon__.getAddonInfo("id");
+__scriptname__ = __addon__.getAddonInfo("name");
+__version__ = __addon__.getAddonInfo("version");
+__language__ = __addon__.getLocalizedString;
+main_url = "http://traduttorianonimi.it";
+
+__cwd__ = xbmc.translatePath(__addon__.getAddonInfo("path")).decode("utf-8");
```
Comment (learningit) [inline service.subtitles.traduttorianonimi/service.py]: In your next update please remove the \__varname\__ style labels. These should only be used for python internals.

**Comment by ShellAddicted** [comment]
is ok now?

File: service.subtitles.traduttorianonimi/resources/lib/TraduttoriAnonimi.py [hunk 3]
```diff
@@ -0,0 +1,157 @@
...
+#  Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston,
+#  MA 02110-1301, USA.
+#  
+# 
+import xbmc;
+class LogStream(object):
+    def write(self,data):
+        xbmc.log(("""*** [TraduttoriAnonimiCore] -> {}""".format(data)).encode('utf-8'), level = xbmc.LOGNOTICE)
+            
+from bs4 import BeautifulSoup;
```
Comment (phil65) [inline service.subtitles.traduttorianonimi/resources/lib/TraduttoriAnonimi.py]: semicolons?

File: service.subtitles.traduttorianonimi/resources/lib/TraduttoriAnonimi.py [hunk 4]
```diff
@@ -0,0 +1,157 @@
...
+        
+    def UpdateShowsList(self):
+        self._GrabShowsListFromWebsite();
+        
+    def _GrabShowsListFromWebsite(self):
+        self.log.info("Grabbing Shows list from Website.")
+        url=self.ShowsListURL
+        self.log.debug("self.ShowsList re-initialized")
+        self.ShowsList=SearchableDict();
+        while (1):
```
Comment (phil65) [inline service.subtitles.traduttorianonimi/resources/lib/TraduttoriAnonimi.py]: This looks rather dangerous. Please limit this to max amount of retries.

File: service.subtitles.traduttorianonimi/service.py [hunk 5]
```diff
@@ -0,0 +1,184 @@
...
+        xbmc.executebuiltin((u"Notification({},{},{})".format(header, msg, time)).encode("utf-8"))
+        
+def RetriveURL(url):
+    try:
+        headers={"user-agent": "Kodi-SubtitleService-TraduttoriAnonimi"};
+        q=requests.get(url,headers=headers);
+        return q;
+    except:
+        log(traceback.format_exc())
+        notify("Errore di rete. controlla la tua connessione",time=3000)
```
Comment (phil65) [inline service.subtitles.traduttorianonimi/service.py]: English, please.

File: service.subtitles.traduttorianonimi/service.py [hunk 6]
```diff
@@ -0,0 +1,184 @@
...
+            for name in q.namelist():
+                if (name.split(".")[-1] in exts):
+                    q.extract(name, temp)
+                    out.append(os.path.join(temp,name))
+            q.close();
+        else:
+            log("Unpacked file detected")
+            try:
+                if (os.path.basename(url).split(".")[-1] in exts):
+                    q=open(os.path.join(temp,os.path.basename(url)),"wb")
```
Comment (phil65) [inline service.subtitles.traduttorianonimi/service.py]: Using "with" statement for opening files would be nice.

File: service.subtitles.traduttorianonimi/service.py [hunk 7]
```diff
@@ -0,0 +1,184 @@
...
+        search(item)
+    
+    elif (params["action"] == "download"):
+        subs=download(params["url"])
+        for sub in subs:
+            listitem = xbmcgui.ListItem(label = sub)
+            xbmcplugin.addDirectoryItem(handle = int(sys.argv[1]), url = sub, listitem = listitem, isFolder = False)
+        xbmcplugin.endOfDirectory(int(sys.argv[1]))
+        
+main()
```
Comment (phil65) [inline service.subtitles.traduttorianonimi/service.py]: newline at EOL, please.

File: service.subtitles.traduttorianonimi/resources/lib/TraduttoriAnonimi.py (hunk 4 above)
Comment (learningit) [inline service.subtitles.traduttorianonimi/resources/lib/TraduttoriAnonimi.py]: Yeah I was wondering about this as well. At a minimum, check for an abort.

Comment (ShellAddicted) [inline service.subtitles.traduttorianonimi/resources/lib/TraduttoriAnonimi.py]: i'm fixing right now. with a (MAX_RETRIES=100 while (MAX_RETRIES>=0)) have you any advice before i commit?

Comment (phil65) [inline service.subtitles.traduttorianonimi/resources/lib/TraduttoriAnonimi.py]: just make it "for in in xrange(3)" or sth like that.

**Comment by ShellAddicted** [comment]
Fixed. is ok now?

File: service.subtitles.traduttorianonimi/addon.xml [hunk 8]
```diff
@@ -0,0 +1,24 @@
+<?xml version="1.0" encoding="UTF-8" standalone="yes"?>
+<addon id="service.subtitles.traduttorianonimi"
+       name="traduttorianonimi.it"
+       version="2.0.2"
+       provider-name="shelladdicted">
+  <requires>
+    <import addon="xbmc.python" version="2.14.0"/>
+    <import addon="script.module.requests" version="2.9.1" />
+    <import addon="script.module.beautifulsoup4" />
```
Comment (phil65) [inline service.subtitles.traduttorianonimi/addon.xml]: Please add a version number to this dependency.

**Comment by phil65** [comment]
Please convert the strings.xml files to strings.po.
See
https://github.com/alanwww1/xbmc-xml2po
https://github.com/alanwww1/xbmc-xml2po/downloads

**Comment by phil65** [comment]
That should be all :) Thanks for your patience.

**Comment by ShellAddicted** [comment]
all done. Thanks for your help.

File: service.subtitles.traduttorianonimi/addon.xml [hunk 9]
```diff
@@ -0,0 +1,24 @@
+<?xml version="1.0" encoding="UTF-8" standalone="yes"?>
+<addon id="service.subtitles.traduttorianonimi"
+       name="traduttorianonimi.it"
+       version="2.0.3"
+       provider-name="shelladdicted">
+  <requires>
+    <import addon="xbmc.python" version="2.14.0"/>
```
Comment (phil65) [inline service.subtitles.traduttorianonimi/addon.xml]: sorry, missed another one :) this should be "2.24.0" (jarvis).

**Comment by ShellAddicted** [comment]
now is 2.24.0

**Comment by phil65** [comment]
Thx!

**Pull request merged**